memcached.c

```diff
@@ -3959,9 +3961,16 @@ static void usage(void) {
            "   - ssl_wbuf_size:       size in kilobytes of per-connection SSL output buffer\n"
            "                          (default: %u)\n", settings.ssl_wbuf_size / (1 << 10));
     printf("   - ssl_session_cache:   enable server-side SSL session cache, to support session\n"
-           "                          resumption\n");
+           "                          resumption\n"
+           "   - ssl_min_version:     minimum protocol version to accept, default is 2(TLSv1.2).\n"
+#ifdef HAVE_OPENSSL_111
```

Comment:
Hmm.. another place with an external library that has random strings which need to be updated in two places. Any better options for this? Guessing we just have to update the code when new versions appear and someone has to manually notice this?

Reply:
Oh; we have some asserts that've been added near the docs for when the changed default no longer matches what's in the docs. add one of those at least?

Reply:
`HAVE_OPENSSL_111` is defined manually in `configure.ac` conditionally on the value of `OPENSSL_VERSION_NUMBER` (a constant whose name generally speaking shouldn't change across versions).

I've reverted the changes to `configure.ac` and replaced `HAVE_OPENSSL_111` with an evaluation against `OPENSSL_VERSION_NUMBER` in 1c5ced4b691580c90cfcb10ee10bff6276318a3f. I think this is a bit clearer, though comes at the expense of repeating the 1.1.1 version constants in a few places. Let me know if this addresses your concern?

Reply:
Ah sorry, What I meant is the help (`-h`) text itself. For other help output with defaults we have asserts that check that the default hasn't changed without the documentation getting updated? That doesn't seem to be what you fixed just here.

Reply:
Ah my bad. In any case though I do feel like 1c5ced4b691580c90cfcb10ee10bff6276318a3f (the unrelated change) is good to have.

In 785c51064b5d3d910a4a406a8e55fbb3608bc87e I parameterized `settings.ssl_min_version` in the help text so that the documentation is always consistent with its default value. This looks a bit more similar to how it's done for the other options. Not entirely sure if this is what you were looking for but let me know.

I think the `verify_default("ssl_min_version", settings.ssl_min_version == TLS1_2_VERSION);` accomplishes the assertion you alluded to but correct me if I'm wrong.

This is what the text output looks like now when compiling on 1.1.1:

```
   - ssl_min_version:     minimum protocol version to accept (default: tlsv1.2)
                          valid values are 0(tlsv1.0), 1(tlsv1.1), 2(tlsv1.2), or 3(tlsv1.3).
```

And on 1.1.0:

```
   - ssl_min_version:     minimum protocol version to accept (default: tlsv1.2)
                          valid values are 0(tlsv1.0), 1(tlsv1.1), or 2(tlsv1.2).
```

Reply:
Got it, looks good. If those were in before I'd missed them somehow.